### PR TITLE
1019 add moon to aladin sky chart on target list page

### PR DESCRIPTION
**Pull request by phycodurus**
Sending you this draft PR to keep you in the loop.

### CODE REVIEW

**Comment by phycodurus** [comment]
This is ready for review now.
1. Uses Astropy for Sun, Moon position and Astroplan for Moon illumination
2. Sun, Moon data passed from inclusion tag to template via context
3. Moon icon greyscale is a function of Moon illumination
4. Also refactored Targets to be in a single Aladin catalog (vs. each target in its own catalog)

Looks like this:
image

File: tom_targets/templates/tom_targets/partials/aladin_skymap.html [hunk 1]
```diff
@@ -15,17 +15,89 @@
...
+    }
+    aladin.addCatalog(targetCatalog);
+
+    // extract Moon information from the context
+    const moonRaDeg = {{ moon_ra|safe }};
+    const moonDecDeg = {{ moon_dec|safe }};
+    const moonIllumination = {{ moon_illumination|safe }}.toFixed(3);
+    const moonGreyscale = 255 * moonIllumination;
+
+    // get a unicode representation of the Moon based on illumination fraction
```
Comment (jchate6) [inline tom_targets/templates/tom_targets/partials/aladin_skymap.html:44]: I went ahead and added the proper phase display code here.

**Review by jchate6 — CHANGES_REQUESTED** [review]
Have a few strong opinions about colors and symbol sizes that I think should be changed.

I also went ahead and just added the phase icons rather than making another 0 priority story to be lost in the icebox for years.

File: tom_targets/templates/tom_targets/partials/aladin_skymap.html [hunk 2]
```diff
@@ -15,17 +15,89 @@
...
+        canvasCtx.fillText(unicodeMoon, source.x, source.y);
+    };
+
+    // create a catalog for the moon
+    const moonImage = A.catalog({shape: drawMoon});
+
+    const popupMoonDescription = `Illumination: ${moonIllumination} <br> RA: ${moonRaDeg.toFixed(4)} <br> Dec: ${moonDecDeg.toFixed(4)}`;
+
+    moonImage.addSources([A.marker(moonRaDeg, moonDecDeg,
+        {popupTitle: 'Geocentric Moon',
```
Comment (jchate6) [inline tom_targets/templates/tom_targets/partials/aladin_skymap.html:76]: This should probably be "Moon (Geocentric)"

File: tom_targets/templates/tom_targets/partials/aladin_skymap.html [hunk 3]
```diff
@@ -15,17 +15,89 @@
...
+        color: 'yellow',
+        sourceSize: 15});
+
+    const sunRaDeg = {{ sun_ra|safe }};
+    const sunDecDeg = {{ sun_dec|safe }};
+
+    const popupSunDescription = `RA: ${sunRaDeg.toFixed(4)} <br> Dec: ${sunDecDeg.toFixed(4)}`;
+
+    sunCatalog.addSources([A.marker(sunRaDeg, sunDecDeg,
+        {popupTitle: 'Geocentric Sun',
```
Comment (jchate6) [inline tom_targets/templates/tom_targets/partials/aladin_skymap.html:95]: Same as for moon.

**Review by jchate6 — APPROVED** [review]
Looks great!

File: tom_targets/templates/tom_targets/partials/aladin_skymap.html [hunk 4]
```diff
@@ -15,17 +15,87 @@
...
 
+    // define catalog for the targets
+    var catalogOptions = {
+        name: 'Targets',
+        color: 'blue',
+        sourceSize: 16};
+    var targetCatalog = A.catalog(catalogOptions);
+
+    // add targets to the catalog
     for (var i = 0; i < targets.length; i++) {
```
Comment (jchate6) [inline tom_targets/templates/tom_targets/partials/aladin_skymap.html:31]: This loop was important for letting people display or remove individual targets.

File: tom_targets/templates/tom_targets/partials/aladin_skymap.html [hunk 5]
```diff
@@ -15,17 +15,87 @@
...
+        canvasCtx.globalAlpha = 1;
+        canvasCtx.font = '25px Arial';
+        canvasCtx.fillStyle = '#eee';
+        canvasCtx.textBaseline = 'middle';
+        canvasCtx.textAlign = 'center';
+        canvasCtx.fillText(unicodeMoon, source.x, source.y);
+    };
+
+    // create a catalog for the moon
+    const moonImage = A.catalog({shape: drawMoon});
```
Comment (jchate6) [inline tom_targets/templates/tom_targets/partials/aladin_skymap.html:69]: This should have a `name: 'Moon'` in the dictionary. My bad.

Comment (jchate6) [reply tom_targets/templates/tom_targets/partials/aladin_skymap.html:69]: Also, can we add a `color: 'gray'` so the catalog tag in the stack menu is vaguely moon colored? It won't do anything to the moon image, but will make it distinct from the many targets.

**Review by jchate6 — CHANGES_REQUESTED** [review]
The things we discussed. Let me know if you'd like me to make these changes.

**Review by jchate6 — APPROVED** [review]

**Pull request merged**